### PR TITLE
feat(issue-list): auto-compact when table exceeds terminal height

### PR DESCRIPTION
**Pull request by BYK**
Auto-detect compact mode based on terminal height. When the estimated
non-compact table height exceeds `process.stdout.rows`, compact (single-line)
rows activate automatically — no flag needed.

### Changes

- **`--compact` is now tri-state**: `--compact` forces ON, `--no-compact` forces
  OFF, omitted → auto-detect via terminal height
- **`shouldAutoCompact(rowCount)`** in `src/lib/formatters/human.ts`: estimates
  table height as `rows × 3 + 4` (2-line content + separator per row, plus
  border/header overhead). Returns `false` for non-TTY to preserve full output
  for piped/scripted usage.
- **`resolveCompact(flag, rowCount)`** in the list command resolves the tri-state
  before passing to `writeIssueTable()`

### Behavior

| Invocation | Result |
|---|---|
| `sentry issue list` | Auto: compact if table won't fit terminal |
| `sentry issue list --compact` | Always compact |
| `sentry issue list --no-compact` | Always full 2-line rows |
| `sentry issue list \| jq` | Full output (non-TTY) |

### Tests

7 property-based tests in `test/lib/formatters/auto-compact.property.test.ts`
covering non-TTY fallback, edge cases, monotonicity, and determinism.

### CODE REVIEW

File: src/lib/formatters/human.ts [hunk 1]
```diff
@@ -393,6 +393,34 @@ function computeAliasShorthand(shortId: string, projectAlias?: string): string {
...
+ * @returns Whether compact mode should be used
+ */
+export function shouldAutoCompact(rowCount: number): boolean {
+  const termHeight = process.stdout.rows;
+  if (!termHeight) {
+    return false;
+  }
+  const estimatedHeight =
+    rowCount * LINES_PER_DEFAULT_ROW + TABLE_LINE_OVERHEAD;
+  return estimatedHeight > termHeight;
```
[the remark this answers is not in the document]
Comment (BYK) [reply src/lib/formatters/human.ts:428]: Good catch! The last data row has no trailing separator (`r > 0 && r < allRows.length - 1` excludes the last row), so the actual height is `3N + 3`, not `3N + 4`. Fixed in 87a0e108 — `TABLE_LINE_OVERHEAD` is now `3`.
